src/App.vue

```diff
@@ -402,18 +403,32 @@ export default {
         this.fetchBusinessInfo(this.businessId),
         this.fetchTasks(this.businessId),
         this.fetchFilings(this.businessId || this.tempRegNumber),
-        this.fetchAddresses(this.businessId),
         this.fetchParties(this.businessId)
       ])
 
-      if (!data || data.length !== 6) throw new Error('Incomplete business data')
+      if (!data || data.length !== 5) throw new Error('Incomplete business data')
 
       this.storeEntityInfo(data[0])
       await this.storeBusinessInfo(data[1])
       this.storeTasks(data[2])
       this.storeFilings(data[3])
-      this.storeAddresses(data[4])
-      this.storeParties(data[5])
+      this.storeParties(data[4])
+    },
+
+    async fetchAddressData ():Promise<void> {
+      let hasBAError = false
+      const data = await Promise.resolve(
+        this.fetchAddresses(this.businessId)
```

Comment:
I think you can simplify this to:
```
const data = this.fetchAddresses(this.businessId)
   .catch(error => {
...
```

Reply:
I think it's because this method is a Promise. When I use .catch directly, it will not work.

Reply:
OK, I'll look at this later.